lib/externalApis/tenderdash/TransactionsClient.js

```diff
@@ -0,0 +1,83 @@
+const EventEmitter = require('events');
+const TransactionWaitPeriodExceededError = require('../../errors/TransactionWaitPeriodExceededError');
+
+const TX_QUERY = 'tm.event = \'Tx\'';
+
+class TransactionsClient extends EventEmitter {
+  /**
+   * @param {WsClient} tenderdashWsClient
+   */
+  constructor(tenderdashWsClient) {
+    super();
+    this.wsClient = tenderdashWsClient;
+    this.wsClient.subscribe(TX_QUERY);
+    this.wsClient.on(TX_QUERY, (message) => {
```

Comment:
Side effect in the constructor is a bad practice. We create an object and already add listeners. It's not clear that it happens and how to remove listeners. Let's move subscribe and listener on the first `waitForTransaction` call?